code/modules/paperwork/photocopier.dm

```diff
@@ -1,223 +1,301 @@
-		else
-			img = image('icons/obj/bureaucracy.dmi', "paper_stamp-dots")
-		img.pixel_x = copy.offset_x[j]
-		img.pixel_y = copy.offset_y[j]
-		c.overlays += img
-	c.updateinfolinks()
-	if(need_toner)
-		toner--
-	if(toner == 0)
-		visible_message("<span class='notice'>A red light on \the [src] flashes, indicating that it is out of toner.</span>")
-	c.update_icon()
-	return c
-
-/obj/machinery/photocopier/proc/photocopy(var/obj/item/photo/photocopy, var/need_toner=1)
-	var/obj/item/photo/p = photocopy.copy()
-	p.dropInto(loc)
-
-	if(toner > 10)	//plenty of toner, go straight greyscale
-		p.img.MapColors(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))//I'm not sure how expensive this is, but given the many limitations of photocopying, it shouldn't be an issue.
-		p.update_icon()
-	else			//not much toner left, lighten the photo
-		p.img.MapColors(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(100,100,100))
-		p.update_icon()
-	if(need_toner)
-		toner -= 5	//photos use a lot of ink!
-	if(toner < 0)
-		toner = 0
-		visible_message("<span class='notice'>A red light on \the [src] flashes, indicating that it is out of toner.</span>")
-
-	return p
-
-//If need_toner is 0, the copies will still be lightened when low on toner, however it will not be prevented from printing. TODO: Implement print queues for fax machines and get rid of need_toner
-/obj/machinery/photocopier/proc/bundlecopy(var/obj/item/paper_bundle/bundle, var/need_toner=1)
-	var/obj/item/paper_bundle/p = new /obj/item/paper_bundle (src)
-	for(var/obj/item/W in bundle.pages)
-		if(toner <= 0 && need_toner)
-			toner = 0
-			visible_message("<span class='notice'>A red light on \the [src] flashes, indicating that it is out of toner.</span>")
-			break
-
-		if(istype(W, /obj/item/paper))
-			W = copy(W)
-		else if(istype(W, /obj/item/photo))
-			W = photocopy(W)
-		W.forceMove(p)
-		p.pages += W
-
-	p.dropInto(loc)
-	p.update_icon()
-	p.icon_state = "paper_words"
-	p.SetName(bundle.name)
-	return p
-
-/obj/item/toner
-	name = "toner cartridge"
-	icon = 'icons/obj/items/tonercartridge.dmi'
-	icon_state = "tonercartridge"
-	material = /decl/material/solid/plastic
-	var/toner_amount = 30
+	if(istype(construct_state, /decl/machine_construction/default/panel_closed) && (istype(O, /obj/item/paper) || istype(O, /obj/item/photo) || istype(O, /obj/item/paper_bundle)))
+		insert_item(O, user)
+		return TRUE
+	return..() //Components attackby will handle refilling with paper and toner
+
+/**Creates a clone of the specified item. Returns a list of cloned items. */
+/obj/machinery/photocopier/proc/scan_item(var/obj/item/I)
+	if(istype(I, /obj/item/paper))
+		var/obj/item/paper/P = I
+		LAZYADD(., P.Clone())
+
+	else if(istype(I, /obj/item/photo))
+		var/obj/item/photo/Ph = I
+		LAZYADD(., Ph.Clone())
+
+	else if(istype(I, /obj/item/paper_bundle))
+		var/obj/item/paper_bundle/B = I
+		for(var/obj/item/page in B.pages)
+			LAZYADD(., scan_item(page))
+
```

Comment:
Another thing that could be handled via the helper or an extension.

Reply:
True